### PR TITLE
style: Introduce `ruff` for linting in all plugins.

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Addresses https://github.com/kedro-org/kedro-plugins/issues/320 and https://github.com/kedro-org/kedro-plugins/issues/331

## Development notes
Replaced `flake8`, `isort` and `pylint` setup with `ruff`.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — APPROVED** [review]
⭐️ Heroic work for removing all the pylint comments

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,178 +14,44 @@ repos:
...
-          - "--select=B,C,E,F,W,T4,B9"
-          - "--ignore=E203,E266,E501,W503"
-        exclude: "^kedro_airflow/dag_template.py|^template.py"
 
   - repo: local
     hooks:
-      # pylint quick checks
-      - id: pylint-quick-kedro-datasets
-        name: "Quick PyLint on kedro_datasets/*"
+      - id: ruff-kedro-datasets
```
Comment (noklam) [inline .pre-commit-config.yaml:20]: https://github.com/kedro-org/kedro/blob/af2ba0ed7abaf004da45a443a0b3d37bf11dd30f/.pre-commit-config.yaml#L6C1-L14C69

    - repo: https://github.com/astral-sh/ruff-pre-commit
      # Ruff version.
      rev: v0.0.277


Can we use the ruff pre-commit hook? this way we don't have to install `ruff` locally and ensure CI always work the same as local.

Comment (merelcht) [reply .pre-commit-config.yaml:20]: I don't know if it's possible to run it only for the plugin that has been changed with the pre-commit hook. @SajidAlamQB do you know if that could work? I know you put a lot of thought into this setup.

Comment (SajidAlamQB) [reply .pre-commit-config.yaml:20]: Yes @merelcht is correct, the main issue is that we can't dynamically run hooks based on the specific plugin environment on the CI. The setup will trigger all the defined ruff linting hooks on all plugins, regardless of which plugin has changes.

A workaround would be to create a script that checks for changed plugins, and then conditionally executes ruff on those plugins.

**Review by ankatiyar — APPROVED** [review]
💯

**Pull request merged**